File: VideoStreamingBackend/src/application/use-cases/ListActiveStreams.ts
/**
 * Use case: List all active streams.
 */
import { StreamSession } from '../../domain/entities/StreamSession';
import { StreamManager } from '../../infrastructure/streaming/StreamManager';

export class ListActiveStreamsUseCase {
  constructor(private readonly streamManager: StreamManager) {}

  execute(): { count: number; streams: StreamSession[] } {
    const streams = this.streamManager.getAllSessions();
    return { count: streams.length, streams };
  }
}

File: StreamingBackend/internal/application/usecases/list_active_streams.go
package usecases

import (
	"github.com/vigileye/streaming-backend/internal/domain/entities"
	"github.com/vigileye/streaming-backend/internal/infrastructure/streaming"
)

// ListActiveStreamsUseCase handles listing all active streams.
type ListActiveStreamsUseCase struct {
	streamManager *streaming.StreamManager
}

// NewListActiveStreamsUseCase creates a new ListActiveStreamsUseCase.
func NewListActiveStreamsUseCase(sm *streaming.StreamManager) *ListActiveStreamsUseCase {
	return &ListActiveStreamsUseCase{streamManager: sm}
}

// Execute returns all active streams.
func (uc *ListActiveStreamsUseCase) Execute() (int, []*entities.StreamSession) {
	streams := uc.streamManager.GetAllSessions()
	return len(streams), streams
}
